### ISSUE TITLE
Cloud storage no longer works on Firefox

### ISSUE THREAD

**Issue by ipkpjersi** [uBlockOrigin/uMatrix-issues#286]
### Prerequisites

- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
    - [x] I also searched the existing issues at <https://github.com/gorhill/uMatrix/issues>
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uMatrix](https://old.reddit.com/r/uMatrix/)
- I tried to reproduce the issue when...
    - [ ] uMatrix extension is wholly disabled or not installed
    - [ ] uMatrix is the only extension
    - [ ] uMatrix with default lists/settings
    - [ ] using a new, unmodified browser profile
- [x] I am running the latest version of uMatrix
- [x] I checked the [documentation](https://github.com/gorhill/uMatrix/wiki) to understand that the issue I report is not a normal behavior
- [ ] I used the logger to rule out that the issue is caused by my ruleset

### Description

The cloud sync feature no longer works on Firefox (I am not sure if it works on Chrome etc), the PC name and updated date are now replaced with "..." and uploading no longer works either. This bug is happepning on multiple PCs.

### Steps to Reproduce

1. Go to dashboard
2. Go to My rules
3. [and so on...]

### Supporting evidence

![](https://i.imgur.com/ri80ep8.png)

### Your environment

* uMatrix version: uMatrix 1.4.0
* Browser Name and version: Firefox 79.0 Snap
* Operating System and version: Ubuntu 18.04 LTS

**Comment by ghost** [comment]
>the PC name and updated date are now replaced with "..."

Pressing on upload button should fill all fields.

When this broke? On Firefox 79? Is this only on Snap? I think Firefox Sync internal storage format changed recently, maybe Snap version does not support it fully.

**Comment by ipkpjersi** [comment]
I did press upload but nothing happens, the fields are still "..."

This broke recently, within the past week, I believe it was still working when Firefox 79 released so I think this might be a new bug?

**Comment by ghost** [comment]
Do you see any errors in Console tab under `about:devtools-toolbox?type=extension&id=uMatrix%40raymondhill.net` address?

**Comment by ipkpjersi** [comment]
No errors of any kind in Console tab.

**Comment by gorhill** [comment]
Maybe you have too many rules, cloud storage maxes at ~100KB. See <https://www.reddit.com/r/uBlockOrigin/comments/i6e7lr/cloud_sync_not_uploading_rules_list/>. Firefox started to enforce quota limit in version 79.

**Comment by ipkpjersi** [comment]
Yep, looks like I ran into the 100KB limit. 

I'm not quite sure I understand - is the 100KB limit a limit of Firefox itself, or a limit of uMatrix?

**Comment by gorhill** [comment]
It's a Firefox limit: <https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/sync#Storage_quotas_for_sync_data>.

**Comment by ipkpjersi** [comment]
Good to know, thanks. I think this issue can now be closed.

**Comment by ghost** [comment]
Can you make it work when you reduce amount of rules? User here https://www.reddit.com/r/uMatrix/comments/inrdrq/ says that it does not work. Maybe it's similar to recently fixed uBO issues?

**Comment by gorhill** [comment]
I stumbled myself onto this issue in uBO/Firefox. I don't use Firefox Sync, only for testing purpose, so it had been a long time since I tried in my Firefox account, and when I did, I found that I had over 400KB of data in _"My filters"_'s cloud storage, and I was now unable to overwrite regardless of what I tried to send to it. The issue was that I had to delete all the storage slots beyond allowed quota.

Now a problem here is that I no longer have time to work on uMatrix. I even can't find the time and motivation to merge pull requests.
